### PR TITLE
Create a @eth-optimism/configs package #6629

### LINKED ISSUE #6629: feat: Create a @eth-optimism/configs package
## Affected Packages

All TS packages

## Problem

- Can't reuse our base configs in other repos
- Caches like nx cache are more complicated to configure to break when configs change
- Root of repo is pretty noisy with configs
- It's generally a bad practice in package based repos to do absolute paths to outside of the package

## Proposed solution

1. Create a new package `@eth-optimism/configs` in `packages/configs`
2. Move tsconfig, prettier config, and eslint config to this package
3. Update references to configs from `../../tsconfig.json` to `@eth-optimism/configs/tsconfig.json`
4. Clean up: Remove the configs from nx.json
5. Clean up: remove the configs from `checked-changed` python script

### PR DESCRIPTION
**Pull request by aliaaquib**
**Description**

A clear and concise description of the features you're adding in this pull request.

- Create a new package `@eth-optimism/configs` in `packages/configs`
- Move` tsconfig, prettier config, and eslint config` to this package
- Update references to `configs` from `../../tsconfig.json` to `@eth-optimism/configs/tsconfig.json`
- Clean up: Remove the `configs` from `nx.json`
- Clean up: remove the `configs` from `checked-changed` python script
**Tests**

Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.

**Additional context**

Add any other context about the problem you're solving.

if there any error or any issue occur  pls let me know 

**Metadata**

- Fixes #6629

### CODE REVIEW

**Review by roninjin10 — CHANGES_REQUESTED** [review]
Hey @aliaaquib, thanks for your contribution! Before we can fully review this, please ensure taht:

- The package is in the right place as described in the issue
- The extends match what's described in the issue.

Please also ensure that `pnpm build` and `pnpm lint` run successfully.

Let me know when you're ready for re-review. Thanks!

**Comment by roninjin10** [comment]
Hey @aliaaquib, anything I can help with to get this pr over the finish line? It's very close

File: packages/@eth-optimism/configs/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (roninjin10) [inline packages/@eth-optimism/configs/tsconfig.json:1]: this file needs to be in packages/configs not packages/@eth-optimism/configs

File: packages/chain-mon/.prettierrc.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module.exports = {
-  ...require('../../.prettierrc.js'),
+  ...require('../@eth-optimism/configs/.prettierrc.js'),
```
Comment (roninjin10) [inline packages/chain-mon/.prettierrc.js:2]:
```suggestion
  ...require('@eth-optimism/configs/.prettierrc.js'),
```
These relative imports all need to be fixed

File: packages/@eth-optimism/configs/tsconfig.json (hunk 1 above)
Comment (aliaaquib) [reply packages/@eth-optimism/configs/tsconfig.json:1]: there is no file configs in packages. can i make configs and add tsconfig.json to it.

Comment (roninjin10) [reply packages/@eth-optimism/configs/tsconfig.json:1]: Yes exactly! The goal of this pr is to make the configs package and put the configs in it. Exactly as the first line of your pr descriptions says: 

> Create a new package @eth-optimism/configs in packages/configs

**Comment by roninjin10** [comment]
Hey @aliaaquib this pr is getting stale. Let me know if there is anything we can do to help get this over the finish line as it's very close

**Pull request closed without merging**